addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3431,9 +3431,23 @@ export class OdooEditor extends EventTarget {
         if (!rangeContent.hasChildNodes()) {
             return;
         }
+        const closestLi = closestElement(selection.anchorNode, 'li');
+        const commonAncestorElement = closestElement(range.commonAncestorContainer);
+        if (commonAncestorElement && !isBlock(rangeContent.firstChild)) {
+            // Get the list of ancestor elements starting from the provided
+            // commonAncestorElement up to the closest li or block-level element.
+            const blockOrLi = closestLi || closestBlock(commonAncestorElement);
+            const ancestorsList = [commonAncestorElement, ...ancestors(commonAncestorElement, blockOrLi)];
+            // Wrap rangeContent with clones of their ancestors to keep the styles.
+            for (const ancestor of ancestorsList) {
+                const clone = ancestor.cloneNode();
+                clone.append(...rangeContent.childNodes);
+                rangeContent.appendChild(clone);
+            }
+        }
         // Repair the copied range.
         if (rangeContent.firstChild.nodeName === 'LI') {
-            const list = range.commonAncestorContainer.cloneNode();
+            const list = closestLi.parentElement.cloneNode();
```

Comment:
Unless I am missing something, we should not change this behavior. This would lead to putting a `<ul>` in the clipboard when copying a single list item, and we don't want that. See comment about adding tests for lists.

Reply:
I have removed this specific changes, and it is functioning properly